docs/features/string-literals-data-section.md

```diff
@@ -0,0 +1,253 @@
+
+### AOT
+
+Ahead-of-time compilation tools would need to be updated to recognize this new pattern of string literal use in place of `ldstr`.
+
+### Automatic threshold
+
+The threshold could be determined automatically with some objective, for example,
+use the utf8 encoding emit strategy for the lowest number of string literals necessary to avoid overflowing the UserString heap.
+
+The set of string literals is not known up front in the compiler, it is discovered lazily (and in parallel) by the emit layer.
+However, we could continue emitting `ldstr` instructions and fix them up in a separate phase after we have seen all the string literals
+(and hence can determine the automatic threshold).
+The `ldstr` and `ldsfld` instructions have the same size, so fixup would be a straightforward replace.
+This fixup phase already exists in the compiler in `MetadataWriter.WriteInstructions`
+(it's also how the string literals are emitted into the UserString heap).
+It is called from `SerializeMethodBodies` which precedes `PopulateSystemTables` call,
+hence synthesizing the utf8 string classes in the former should be possible and they would be emitted in the latter.
+
+### Statistics
+
+The compiler could emit an info diagnostic with useful statistics for customers to determine what threshold to set.
+For example, a histogram of string literal lengths encountered while emitting the program.
+Tools like ILSpy can already be used to obtain some related data, e.g., count, length, and contents of string literals in the UserString heap.
+
+### Single blob
+
+We could merge all UTF-8 byte sequences for affected strings into a single blob - a virtual User-String heap.
+The sequences can be deduped as we dedupe entries in User-String heap.
+We can even attempt to do some vary basic compression,
+for example, in the name heap that we generate (contains type and member names, etc.), names with shared content can overlap.
+We would generate a single `__StaticArrayInitTypeSize=*` structure for the entire virtual heap blob and
+add a single `.data` field to `<PrivateImplementationDetails>` that points to the blob.
+At runtime, we would do an offset to where the required data reside in the blob and decode the required length from UTF-8 to UTF-16.
+
+## Alternatives
+
+### Configuration/emit granularity
+
+Instead of one global feature flag, the emit strategy could be controlled using compiler-recognized attributes (applicable to assemblies or classes).
+Furthermore, we could emit more than one string per one class. That could be configurable as well.
+
+One interesting strategy would be grouping strings which occur in one user-defined type into one compiler-generated `<S>` type.
+The idea is that strings from one class are likely used "together" so there is no performance impact from initializing them all at once and we save on metadata.
+
+### GC
+
+To avoid rooting the `string` references forever, we could turn the fields into `WeakReference<string>`s.
```

Comment:
This would provide no benefit and it would be quite expensive - for both direct overhead and indirectly for the GC (longer GC pause times). You can scratch this idea.